src/Cake.Common/Build/GitHubActions/Commands/GitHubActionsCommands.cs

```diff
@@ -103,6 +103,28 @@ public void SetEnvironmentVariable(string key, string value)
             writer.WriteLine("CAKEEOF");
         }
 
+        /// <summary>
+        /// Creates or updates the step summary for a GitHub workflow.
+        /// </summary>
+        /// <param name="summary">The step summary.</param>
+        public void SetStepSummary(string summary)
+        {
+            if (string.IsNullOrEmpty(summary))
+            {
+                throw new ArgumentNullException(nameof(summary));
+            }
+
+            if (_actionsEnvironment.Runtime.StepSummary == null)
+            {
+                throw new CakeException("GitHub Actions Runtime StepSummary missing.");
+            }
+
+            var file = _fileSystem.GetFile(_actionsEnvironment.Runtime.StepSummary);
+            using var stream = file.Open(FileMode.Append, FileAccess.Write, FileShare.None);
+            using var writer = new StreamWriter(stream);
+            writer.Write(summary);
```

Comment:
This should probably use `WriteLine` instead, which also relates to the echo example in [multiline markdown content](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#multiline-markdown-content)?

Reply:
Agreed, I changed this in the rebase.